tests/collections/tts/modules/test_audio_codec_modules.py

```diff
@@ -205,3 +207,183 @@ def test_snake(self):
         snake = CodecActivation('snake', channels=self.in_channels)
         out = snake(x=inputs)
         assert out.shape == (self.batch_size, self.in_channels, self.max_len)
+
+
+class TestFiniteScalarQuantizer:
+    def setup_class(self):
+        """Setup common members
+        """
+        self.batch_size = 2
+        self.max_len = 20
+        self.num_examples = 10
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize('num_levels', [[2, 3], [8, 5, 5]])
+    def test_fsq_eval(self, num_levels: list):
+        """Simple test to confirm that the FSQ module can be instantiated and run,
+        and that forward produces the same result as encode-decode.
+        """
+        fsq = FiniteScalarQuantizer(num_levels=num_levels)
+
+        for i in range(self.num_examples):
+            inputs = torch.randn([self.batch_size, fsq.codebook_dim, self.max_len])
+            input_len = torch.tensor([self.max_len] * self.batch_size, dtype=torch.int32)
+
+            # apply forward
+            dequantized_fw, indices_fw = fsq(inputs=inputs, input_len=input_len)
+
+            assert dequantized_fw.max() <= 1.0, f'example {i}: dequantized_fw.max() is {dequantized_fw.max()}'
+            assert dequantized_fw.min() >= -1.0, f'example {i}: dequantized_fw.min() is {dequantized_fw.min()}'
+
+            # encode-decode
+            indices_enc = fsq.encode(inputs=inputs, input_len=input_len)
+            dequantized_dec = fsq.decode(indices=indices_enc, input_len=input_len)
+
+            # make sure the results are the same
+            torch.testing.assert_close(indices_enc, indices_fw, msg=f'example {i}: indices mismatch')
+            torch.testing.assert_close(dequantized_dec, dequantized_fw, msg=f'example {i}: dequantized mismatch')
+
+    @pytest.mark.unit
+    def test_fsq_output(self):
+        """Simple test to make sure the output of FSQ is correct
+        for a single setup.
+        """
+        num_levels = [2, 3]
+        fsq = FiniteScalarQuantizer(num_levels=num_levels)
+
+        # # To generate inputs & outputs for testing
+        # max_len = 8
+        # inputs = torch.randn([self.num_examples, fsq.codebook_dim, max_len])
+        # input_len = torch.tensor([max_len] * self.num_examples, dtype=torch.int32)
+        # dequantized, indices = fsq(inputs=inputs, input_len=input_len)
+        # print(inputs)
+        # print(input_len)
+        # print(dequantized)
+        # print(indices)
+
+        # inputs
+        inputs = torch.tensor(
+            [
+                [
+                    [0.6572, 1.3574, 2.1646, 0.9457, -0.3489, 0.6732, -0.7148, -0.5143],
+                    [0.5123, 0.8552, 1.7814, 1.9938, -1.1909, -0.9991, -3.7932, -0.4438],
+                ],
+                [
+                    [0.2357, 0.8324, 0.8932, -0.0596, 0.6130, -0.0299, 0.3824, 1.6278],
+                    [-0.3781, 0.1864, -0.2190, 1.2199, -1.1398, -0.8443, -0.7865, 0.1470],
+                ],
```

Comment:
Should we use smaller dimension inputs/outputs for the tests? 80 elements looks unnecessarily big for a reader to try and parse and validate for correctness.